README.md

```diff
@@ -1,2 +1,138 @@
+[![Jenkins Plugin Installs](https://img.shields.io/jenkins/plugin/i/docker-custom-build-environment.svg?color=blue)](https://plugins.jenkins.io/docker-custom-build-environment/)
+
+This plugin allows the definition of a build environment for a job using a Docker container.
+
+<img src="docs/images/docker-jenkins.png" width="300">
+
+## Introduction
+
+A major requirements for a Jenkins-based Continuous Integration/Delivery
+setup are:
+
+- Ensure the build/test environment is exactly reproducible from a pre-defined setup.
+- Ensure the build/test environment is well isolated from other builds executed on the same infrastructure.
+
+Docker is a great way to bootstrap reproducible and isolated environments. Compared to a virtual machine, it's faster to launch and lighter to use. Another benefit is the Docker image can be defined both as a binary image pulled from repository, or as a plain text Dockerfile you can store in SCM along side the project source code, so that the source and environment are always in sync and recorded.
+
+CloudBees Docker Custom Build Environment Plugin has been designed to make Docker Images / [Dockerfile](https://docs.docker.com/reference/builder/) a first class citizen in a continuous delivery setup, which allows the simplest way for a development team to manage the exact bits of the build environment whilst the infrastructure team only has to focus on available resources hosting arbitrary docker containers.
+
+## User Guide
+
+CloudBees Docker Custom Build Environment Plugin can be used from any job type, it appears in the "build environment" section and let you configure your build to run inside a Docker container.
+
+You can :
+
+- Select the Docker image to run the build as a **Docker image** to be pulled. This is comparable to the [docker-plugin](https://wiki.jenkins.io/display/JENKINS/Docker+Plugin) approach to offer docker agents within Jenkins, but without any prerequisites on the Docker image nor need for Administrator privileges to configure the adequate agent template.
+
+- Configure the plugin to build a container image **from a  Dockerfile** stored in project repository. With this setup, you get both the project source code and build environment defined in SCM. This is my preferred way to use this plugin.
+
+SCM checkout will run within a classic Jenkins agent execution context - this is required to access the project Dockerfile then build and run the required container.
+
+### Using Docker image
+
+CloudBees Docker Custom Build Environment let you use arbitrary docker image to host the build. You can use such an image you build on your own (or using [CloudBees Docker Build and Publish](https://wiki.jenkins.io/display/JENKINS/Docker+build+publish+Plugin) plugin) to define the set of prerequisites for your project and share with the development team, as well as reuse for your CI job. In
```

Comment:
Prefer to link to the plugins site rather than the wiki.
```suggestion
CloudBees Docker Custom Build Environment let you use arbitrary docker image to host the build. You can use such an image you build on your own (or using [CloudBees Docker Build and Publish](https://plugins.jenkins.io/docker-build-publish/) plugin) to define the set of prerequisites for your project and share with the development team, as well as reuse for your CI job. In
```